src/include/migraphx/matcher.hpp

```diff
@@ -896,10 +897,20 @@ inline auto has_value(T x, std::size_t atol_mult = 10, std::size_t rtol_mult = 1
             else
             {
                 auto eps = std::numeric_limits<type>::epsilon();
-                if(std::all_of(v.begin(), v.end(), [&](auto val) {
-                       return std::fabs(val - static_cast<type>(x)) <
-                              eps * (atol_mult + rtol_mult * std::fabs(val));
-                   }))
+                auto tolerance = atol_mult + rtol_mult * std::fabs(x);
+                if(migraphx::float_equal(tolerance, 0.0))
```

Comment:
Confusing logic. `atol_mult== 0 && rtol_mult == 0` is all that should be checked here on line 901 -- and tolerance (or its equivalent) shouldn't be checked for 0.0 here, as that gives the impression that somehow atol_mult == -1 * rtol_mult * x is a valid use_case to execute this if clause.

Nitpicking: If you want to optimize and save the result of variable `tolerance`, please do by creating an else clause on line 911. Also, in that case, move variable `eps` into that else clause -- better combine tolerance and eps.

Reply:
> atol_mult == -1  
Atol is std::size_t it can't be -1

Reply:
Well, the code and the intent should match, really that is the point. You can of course ignore this suggestion, and I am perfectly fine with it. Thanks.